centreon/src/Core/HostCategory/Infrastructure/Repository/DbReadHostCategoryRepository.php

```diff
@@ -198,4 +198,71 @@ private function hasRestrictedAccessToHostCategories(array $accessGroupIds): boo
 
         return (bool) $statement->fetchColumn();
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function exists(int $hostCategoryId): bool
+    {
+        $this->info('Check existance of host category with id #' . $hostCategoryId);
+
+        $request = $this->translateDbName(
+            'SELECT hc.hc_id FROM `:db`.hostcategories hc WHERE hc.hc_id = :hostCategoryId'
```

Comment:
As told by Laurent on my own PRs, there a _relative_ risk around the `hc_id=0` which will be cast to `false`.
Prefer then the use of `SELECT 1` which is `hc_id` agnostic.
```suggestion
            'SELECT 1 FROM `:db`.hostcategories hc WHERE hc.hc_id = :hostCategoryId'
```